src/core/services/theming/colors.js

```diff
@@ -0,0 +1,129 @@
+            var colors  = [];
+            colorStore[paletteName] = colors;
+            Object.keys(palette).forEach(copyColors);
+            return ;
+
+            function copyColors(colorName) {
+                // use an regex to look for hex colors, ignore the rest
+                if (/#[0-9A-Fa-f]{6}|0-9A-Fa-f]{8}\b/.exec(palette[colorName])) {
+                    colors.push({color:colorName,value:palette[colorName]});
+                }
+            }
+        }
+    }
+
+    loadDefaults.$inject=['$mdColors'];
+    function loadDefaults($mdColors) {
+        // this sets the default that is stored in the config-fase into the service!
+        $mdColors.palettes = Object.keys(colorStore);
+        $mdColors.loadPalette(defaultPalette);
+    }
+
+    mdColors.$inject = ['$interpolate'];
+    function mdColors ($interpolate) {
+        // wrap all of the above up in a reusable service.
+        var service        = this;
+        service.palette    = defaultPalette;
+        service.colors     = [];
+        service.palettes   = [];
+        service.loadPalette = loadPalette;
+
+        return service;
+
+        function loadPalette(newPalette) {
+            if(service.palettes.indexOf(newPalette) === -1) {
+                throw new Error('$mdColors.loadPalette: Unknown palette name ' + newPalette);
+            }
+            service.palette = newPalette;
+            service.colors  = colorStore[newPalette];
+            createStyleSheet();
+        }
+
+        function createStyleSheet () {
+            var colors = service.colors;
+            var fg, bg;
+            if (typeof customSheet === 'undefined') {
+                // use closure for caching the styleSheet
+                newStyleSheet();
+            } else {
+                // remove existing rules
+                // TODO: look into disabling/enabling pre-build style-guides
+                //       in stead of delete and recreate!
+                while (customSheet.cssRules.length > 0 ) {
+                    customSheet.deleteRule(0);
+                }
+            }
+
+            // set up interpolation functions to build css rules.
+            fg  = $interpolate('.md-fg-{{color}} { color:{{value}};}');
+            bg  = $interpolate('.md-bg-{{color}} { background-color:{{value}};}');
+
+            colors.forEach(function (color) {
+                // insert foreground color rule
+                customSheet.insertRule(fg(color));
+                // insert background color rule
+                customSheet.insertRule(bg(color));
+            });
+        }
+    }
+
+    function newStyleSheet() {
+        // function to ad an dynamic style-sheet to the document
+        var style   = document.createElement("style");
+        style.title = 'Dynamic Generated my Angular-Material';
+        // WebKit hack... (not sure if still needed)
+        style.appendChild(document.createTextNode(""));
+
+        document.head.appendChild(style);
```

Comment:
Please don't use `document` directly :smiley:
Use it via an DI-ed service instead (`$window`, `$document`).